docs-src/docs/guides/terraform_beyond_layer0.md

```diff
@@ -0,0 +1,158 @@
+
+State path: terraform.tfstate
+
+Outputs:
+
+guestbook_url = <http endpoint for the sample application>
+```
+
+!!! Note
+  It may take a few minutes for the guestbook service to launch and the load balancer to become available. During that time you may get HTTP 503 errors when making HTTP requests against the load balancer URL.
+
+Terraform will set up the entire environment for you and then output a link to the application's load balancer.
+
+### What's happening
+Terraform using the AWS provider, provisions a new DynamoDB Table (with the name you have configured) and configures environment variables for the Layer0 application to use the newly provisioned table, and deploys the application into a Layer0 environment using the Layer0 provider.
+
+Looking at an excerpt of [layer0.tf](https://github.com/quintilesims/layer0-examples/blob/master/guestbook-db/layer0.tf) file, we can see the following definitions:
+
+```
+resource "aws_DynamoDB_table" "guestbook" {
+  name           = "${var.table_name}"
+  read_capacity  = 20
+  write_capacity = 20
+  hash_key       = "id"
+
+  attribute {
+    name = "id"
+    type = "S"
+  }
+}
+
+resource "layer0_deploy" "guestbook" {
+  name    = "guestbook"
+  content = "${data.template_file.guestbook.rendered}"
+}
+
+data "template_file" "guestbook" {
+  template = "${file("Dockerrun.aws.json")}"
+
+  vars {
+    access_key = "${var.access_key}"
+    secret_key = "${var.secret_key}"
+    region     = "${var.region}"
+    table_name = "${aws_DynamoDB_table.guestbook.name}"
+  }
+}
+```
+
+Note the resource definitions for `aws_dynamodb_table` and `layer0_deploy`. To configure the guestbook application to use the provisioned DynamoDB table, we reference the `name` property from the DynamoDB definition `table_name = "${aws_dynamodb_table.guestbook.name}"`. 
+
+This is then used to populate the template fields in our [Dockerrun.aws.json](https://github.com/quintilesims/layer0-examples/blob/master/guestbook-db/Dockerrun.aws.json) file. 
+
+```
+{
+    "AWSEBDockerrunVersion": 2,
+    "containerDefinitions": [
+        {
+            "name": "guestbook",
+            "image": "quintilesims/guestbook-db",
+            "essential": true,
+            "memory": 128,
+            "environment": [
+                {
+                    "name": "DYNAMO_TABLE",
+                    "value": "${table_name}"
+                }
+                ...
+```
+
+The Layer0 configuration referencing the AWS DynamoDB configuration `table_name = "${aws_DynamoDB_table.guestbook.name}"`, infers an implicit dependency. Before Terraform creates the infrastructre, it will use this information to order the resources created and also create resources in parallel where there are no dependencies. In this example, the AWS DynamoDB table will be created before the Layer0 deploy. See [Terraform Dependencies](https://www.terraform.io/intro/getting-started/dependencies.html) for more information.
+
+## Part 4: Terraform Destroy
```

Comment:
I think we should add more into this guide. As is, it really isn't adding much more value than the guestbook w/ terraform walkthroughs. I think we should add common workflows:
* Talk about `terraform.tfstate`. Include how to use a [remote backend](https://www.terraform.io/intro/getting-started/remote.html) to allow locking and redundancy. 
* Updating the configuration. Maybe make the scale go from 1 to 3, run `terraform plan` then `terraform apply`.
* Throw in some other best practices with terraform+layer0.